syntax = "proto3";

message NfcConfigMessage {
    optional bool factory = 1;

    optional string secret_key = 2;
    optional uint32 serial_number = 3;
    optional uint32 nonce_counter = 4;

    optional Lorawan lorawan = 5;
    optional Application application = 6;

    message Lorawan {
        enum Region {
            EU868 = 0;
            US915 = 1;
            AU915 = 2;
        }

        enum Activation {
            ABP = 0;
            OTAA = 1;
        }

        enum Network {
            PUBLIC = 0;
            PRIVATE = 1;
        }

        optional Region region = 1;
        optional Network network = 2;
        optional bool adr = 3;
        optional Activation activation = 4;
        optional string deveui = 5;
        optional string joineui = 6;
        optional string nwkkey = 7;
        optional string appkey = 8;
        optional string devaddr = 9;
        optional string nwkskey = 10;
        optional string appskey = 11;
    }

    message Application {
        optional bool calibration = 1;
        optional uint32 interval_sample = 2;
        optional uint32 interval_aggreg = 3;
        optional uint32 interval_report = 4;
        optional float alarm_int_temp_lo = 5;
        optional float alarm_int_temp_hi = 6;
        optional float alarm_int_temp_hst = 7;
        optional float alarm_t1_temp_lo = 8;
        optional float alarm_t1_temp_hi = 9;
        optional float alarm_t1_temp_hst = 10;
        optional float alarm_t2_temp_lo = 11;
        optional float alarm_t2_temp_hi = 12;
        optional float alarm_t2_temp_hst = 13;
        optional bool hall_left_enabled = 14;
        optional bool hall_left_counter = 15;
        optional bool hall_left_notify_act = 16;
        optional bool hall_left_notify_deact = 17;
        optional bool hall_right_enabled = 18;
        optional bool hall_right_counter = 19;
        optional bool hall_right_notify_act = 20;
        optional bool hall_right_notify_deact = 21;
    }
}
